### PR TITLE
fix(#8837): fix NaN in telemetry doc id

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

- Adds regex to check the telemetry name
- Adds warning for telemetry names that don't match the new name format.

Testing: https://github.com/medic/cht-core/assets/66472237/436da61e-ef74-4466-8481-66445beb6932

#8837

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:check-telemetry-db-name/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:check-telemetry-db-name/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:check-telemetry-db-name/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
@garethbowen @ralfudx I ran the v4.2.x and 4.5.x branches, and after several tries, I couldn't get a bad telemetry record. I straightened the validations and added an exception, so next time, we will have more info from the feedback docs. 
Let me know your suggestions. Thanks!

**Comment by garethbowen** [comment]
@latin-panda Thanks for getting right on this!

To do a review I need more information, like how you determined this is the problem, and why this is the right fix. The best place for this is probably a comment on the issue.

**Comment by latin-panda** [comment]
>  how you determined this is the problem

@garethbowen I haven't been able to reproduce it. So maybe @ralfudx can help? 
For now, this is just making the validations stronger by processing the telemetry db names with the correct format. 
I'll add this in the ticket in a couple of minutes

File: webapp/src/ts/services/telemetry.service.ts [hunk 1]
```diff
@@ -345,10 +356,8 @@ export class TelemetryService {
     databaseNames?.forEach(dbName => {
       const nameNoPrefix = dbName?.replace(this.POUCH_PREFIX, '') || '';
 
-      // Skips new Telemetry DB, then matches the old deprecated Telemetry DB.
-      if (!nameNoPrefix.startsWith(this.TELEMETRY_PREFIX)
-        && nameNoPrefix.includes(this.TELEMETRY_PREFIX)
-        && nameNoPrefix.includes(this.sessionService.userCtx().name)) {
```
Comment (garethbowen) [inline webapp/src/ts/services/telemetry.service.ts:351]: Why did you remove the check for `userCtx`? Without this won't we end up deleting someone elses telemetry on phones that are shared between users?

**Comment by garethbowen** [comment]
@latin-panda Sorry I wrote that comment before you updated the PR description and added the comment. I deleted it again once I saw the new details. Sorry for the confusion!

**Review by garethbowen — CHANGES_REQUESTED** [review]
I worry this is going to hide the error, because now any database that doesn't match the name will be deleted. In other words, if this happens again it'll silently fail rather than loudly fail. Because we haven't managed to reproduce this yet, we can't be certain if deletion is the right approach or if there's a bug in how we're creating the dbs in the first place.

Let's put some more time into trying to reproduce it.

File: webapp/src/ts/services/telemetry.service.ts (hunk 1 above)
Comment (latin-panda) [reply webapp/src/ts/services/telemetry.service.ts:351]: ah shared phones, okay putting it back

**Comment by latin-panda** [comment]
@ralfudx and @garethbowen please have a look. The momentjs format was translating the numbers to Nepali, I have updated the ticket with this.  

@ralfudx Please check the PR description, I have added a video showing how to test this error. Let me know any questions or suggestions.

**Review by garethbowen — APPROVED** [review]
LGTM. Thanks for adding the bikram sambat test!

**Review by ralfudx — APPROVED** [review]
LGTM
